pkg/build/builder/sti_test.go

```diff
@@ -0,0 +1,168 @@
+package builder
+
+import (
+	"errors"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/openshift/origin/pkg/build/api"
+	stiapi "github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/api/validation"
+	"github.com/openshift/source-to-image/pkg/build"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"strings"
+	"testing"
+)
+
+// mock docker client
+type testDockerClient struct {
+	// flag if BuildImage was called
+	buildImageCalled bool
+	// flag if PushImage was called
+	pushImageCalled bool
+	// flag if RemoveImage was called
+	removeImageCalled bool
+	// emulated error to return from PushImage call
+	errPushImage error
+}
+
+// Registers a call and returns an error (if any)
+func (client testDockerClient) BuildImage(opts docker.BuildImageOptions) error {
+	return nil
+}
+
+// Registers PushImage call and returns an error (if any)
+func (client testDockerClient) PushImage(opts docker.PushImageOptions, auth docker.AuthConfiguration) error {
+	client.pushImageCalled = true
+	return client.errPushImage
+}
+
+// Registers RemoveImage call and returns an error (if any)
+func (client testDockerClient) RemoveImage(name string) error {
+	return nil
+}
+
+// Mock S2I Builder factory implementation
+type testStiBuilderFactory struct {
+	// error to return from GetStrategy function
+	getStrategyErr error
+	// error to return from Build function
+	buildError error
+}
+
+// Mock S2I Config Validator implementation
+type testStiConfigValidator struct {
+	// errors to return
+	errors []validation.ValidationError
+}
+
+// Mock S2I builder factory implementation. Just returns mock S2I builder instances ot error (if set)
```

Comment:
I'll just remove the godocs from tests, we don't use them anywhere in our codebase.